webapp/components/ContentMenu/ContentMenu.spec.js

```diff
@@ -0,0 +1,543 @@
+        const wrapper = openContentMenu({
+          isOwner: false,
+          resourceType: 'contribution',
+          resource: {
+            id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+            pinnedBy: 'someone',
+          },
+        })
+        let items = wrapper.findAll('.ds-menu-item')
+        // for (var i = 0; i < items.length; i++) { console.log(items.at(i).html()) }
+        expect(items.at(0).text()).toBe('post.menu.unpin')
+        expect(
+          items
+            .at(0)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/post-menu-unpin')
+        items.at(0).trigger('click')
+        expect(wrapper.emitted('unpinPost')).toEqual([
+          [
+            {
+              id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+              pinnedBy: 'someone',
+            },
+          ],
+        ])
+      })
+    })
+
+    describe('comment', () => {
+      it('owner can edit and delete comment', () => {
+        const wrapper = openContentMenu({
+          isOwner: true,
+          resourceType: 'comment',
+          resource: {
+            id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+          },
+        })
+        openModalSpy = jest.spyOn(wrapper.vm, 'openModal')
+        let items = wrapper.findAll('.ds-menu-item')
+        expect(items).toHaveLength(2)
+        expect(items.at(0).text()).toBe('comment.menu.edit')
+        expect(
+          items
+            .at(0)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/comment-menu-edit')
+        items.at(0).trigger('click')
+        expect(wrapper.emitted('showEditCommentMenu')).toEqual([[true]])
+        expect(items.at(1).text()).toBe('comment.menu.delete')
+        expect(
+          items
+            .at(1)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/comment-menu-delete')
+        items.at(1).trigger('click')
+        expect(openModalSpy).toHaveBeenCalledWith('delete')
+      })
+    })
+
+    describe('report', () => {
+      it('anyone who is not owner can report post', () => {
+        getters['auth/isAdmin'] = () => false
+        getters['auth/isModerator'] = () => false
+        const wrapper = openContentMenu({
+          isOwner: false,
+          resourceType: 'contribution',
+          resource: {
+            id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+          },
+        })
+        openModalSpy = jest.spyOn(wrapper.vm, 'openModal')
+        let items = wrapper.findAll('.ds-menu-item')
+        expect(items).toHaveLength(1)
+        expect(items.at(0).text()).toBe('report.contribution.title')
+        expect(
+          items
+            .at(0)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/report-contribution-title')
+        items.at(0).trigger('click')
+        expect(openModalSpy).toHaveBeenCalledWith('report')
```

Comment:
instead of setting up the `method` openModal as a spy, can we set up `$store.commit` as a mock, and check that is called with specific data?

Reply:
I tried, but it is getting very complicated because y also have to mount the Modal component. I think we have to rely on the tests of the Modal component that the commits are called correctly (which as far as I can see is the case). I ensure in the tests of ContentMenu that the parameters are passed to Modal correctly.

Reply:
I can see that, and as we discussed today, you are testing it up to one point, and there are other tests that take it further!!